spec/recipes/setup_shclustering_spec.rb

```diff
@@ -0,0 +1,138 @@
+        node.set['splunk']['is_server'] = true
+        node.set['splunk']['shclustering']['enabled'] = true
+        node.set['splunk']['shclustering']['deployer_url'] = "https://#{deployer_node.fqdn}:8089",
+        node.set['splunk']['shclustering']['mgmt_uri'] = "https://#{node['fqdn']}:8089",
+        node.set['splunk']['shclustering']['shcluster_members'] = [ 
+        	"#{node['splunk']['shclustering']['mgmt_uri']}"
+        ]
+        # Populate mock vault data bag to the server
+        server.create_data_bag('vault', secrets)
+      end.converge(described_recipe)
+    end
+
+    it 'includes chef-vault' do
+      expect(chef_run).to include_recipe('chef-vault::default')
+    end
+
+    it 'writes a file marker to ensure convergence' do
+      expect(chef_run).to render_file('/opt/splunk/etc/.setup_shcluster').with_content('true\n')
+    end
+
+		it 'writes server.conf with replication port' do
+			expect(chef_run).to render_file('/opt/splunk/etc/apps/0_autogen_shcluster_config/local/server.conf')
+			.with_content("[replication_port://#{chef_run.node['splunk']['shclustering']['replication_port']}]")
+		end
+
+		it 'writes server.conf with a shclustering stanza' do
+			expect(chef_run).to render_file('/opt/splunk/etc/apps/0_autogen_shcluster_config/local/server.conf')
+			.with_content("[shclustering]")
+		end
+
+		it 'writes server.conf with the deployer url' do
+			expect(chef_run).to render_file('/opt/splunk/etc/apps/0_autogen_shcluster_config/local/server.conf')
+			.with_content("conf_deploy_fetch_url = #{chef_run.node['splunk']['shclustering']['deployer_url']}")
+		end
+
+		it 'writes server.conf with the node mgmt uri' do
+			expect(chef_run).to render_file('/opt/splunk/etc/apps/0_autogen_shcluster_config/local/server.conf')
+			.with_content("mgmt_uri = #{chef_run.node['splunk']['shclustering']['mgmt_uri']}")
+		end
+
+		it 'writes server.conf with the shcluster replication factor' do
+			expect(chef_run).to render_file('/opt/splunk/etc/apps/0_autogen_shcluster_config/local/server.conf')
+			.with_content("replication_factor = #{chef_run.node['splunk']['shclustering']['replication_factor']}")
+		end
+
+		it 'writes server.conf with the shcluster label' do
+			expect(chef_run).to render_file('/opt/splunk/etc/apps/0_autogen_shcluster_config/local/server.conf')
+			.with_content("shcluster_label = #{chef_run.node['splunk']['shclustering']['label']}")
+		end
+
+		it 'writes server.conf with the shcluster secret' do
+			expect(chef_run).to render_file('/opt/splunk/etc/apps/0_autogen_shcluster_config/local/server.conf')
+			.with_content("pass4SymmKey = #{chef_run.node['splunk']['shclustering']['shcluster_secret']}")
+		end
+
+    context 'while set to captain mode' do
+    	before(:each) do
+	      chef_run.node.set['splunk']['shclustering']['mode'] = "captain"
+	      chef_run.converge(described_recipe)
+	    end
+
+			let(:shcluster_servers_list) do
+				chef_run.node['splunk']['shclustering']['shcluster_members'].join(',')
```

Comment:
Yes. Do you want to an execute another new chef_run? If so, you'd want re-define `:chef_run` similar to https://github.com/chef-cookbooks/chef-splunk/blob/master/spec/recipes/install_server_spec.rb
or https://github.com/chef-cookbooks/chef-splunk/blob/master/spec/recipes/setup_clustering_spec.rb
I'm not sure but it may be the cause of the error you reported below..

Minor suggestion: may want to manually hardcode `shcluster_servers_list` in the `let` statement to ensure test integrity vs setting it based on node attribute.